### ISSUE TITLE
New Drawing Tool - Line Arrow

### ISSUE THREAD

**Issue by shishima123** [amcharts/amcharts5#1709]
Many users want a tool to draw straight lines with arrows. I think with this tool, users will be able to highlight specific positions on the screen. Currently, they are using the line tool with the pencil to draw arrows.

Could you consider adding this tool in the future?

![image](https://github.com/user-attachments/assets/4c61ff12-9417-4d62-a8c9-3edadabeeacc)

**Comment by martynasma** [comment]
Implemented in [5.10.5](https://github.com/amcharts/amcharts5/releases/tag/5.10.5).

## [5.10.5] - 2024-09-06

### Added
- `scrollToChild()` method added to `Container`. Scrolls the container to the target child element if the container is scrollable and the element is currently outside of view.
- New `StockChart` drawing tool added: "Line Arrow".

### Changed
- When a selected drawing in a `StockChart` is inselected by clicking on a chart's plot area the new drawing will not be added anymore.

### Fixed
- Keyboard-dragging of focusable elements was not working correctly in some cases.
- Sometimes label/callout drawing text was cut temporarily while editing them on `StockChart`.
- Editable labels/callouts would go into edit mode when unserialized.
- If auto-saving of drawings was enabled, editing text would not always save the latest version of the label/callout text.
- Dragging a `StockChart` drawing while other drawing was selected, would drag the selected drawing as well.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by shishima123** [comment]
Thank you for adding it @martynasma 

I tried applying it and realized that you added a straight line with an arrow overlapping at the top of the line. This causes an issue when moving the line, where the arrow has a slight delay before overlapping with the line.

This is not a major issue, and users might accept it.

However, when I hide the bullets at both ends, ahead of the arrow there is an extra part of the line
Could you shorten the line a little?

Also, when I clear all drawings using drawingControl.clearDrawings() the arrows still remain on the screen.

Here are the 2 functions I use to clear all drawings and hide bullets.
```js
function clearAllDrawing() {
  drawingControl.clearDrawings()
}

function disableDrawBullet(state) {
  stockChart.panels.each((panel) => {
    panel.series.each((series) => {
      if (
        series.isType('DrawingSeries') &&
        !['LabelSeries', 'IconSeries'].includes(series.className)
      ) {
        series.bulletsContainer.set('forceHidden', state)
      }
    })
  })
}
```

https://github.com/user-attachments/assets/412a0486-f386-41d8-a5eb-ff9de1705a3c

**Comment by martynasma** [comment]
Implemented in [5.10.6](https://github.com/amcharts/amcharts5/releases/tag/5.10.6).

## [5.10.6] - 2024-10-11

### Added
- `hideDrawingGrips` setting (default: `false`) added to `StockChart`. If set to true, grips of drawings will be hidden when not in drawing mode.
- `toggleDrawing(enabled: boolean)` method added to `StockChart`. Toggles drawing mode on or off.
- `autoOpenSettings` setting (default: `true`) added to `Indicator`. If set to `false`, a settings modal will not pop up automatically when the indicator is added to chart via indicator control.

### Changed
- On `StockChart` labels and icons are no longer draggable to avoid interfering with chart drag/zoom.
- Arrow position slightly changed in "Line Arrow" drawing tool on `StockChart`. Grip at the end of the arrow removed.
- Arrow tip size will not increased to reflect the line thickness of a "Line Arrow".
- "Volume Profile" indicator had its full name hardcoded for legend. It will now use its `shortName` (which is "Volume Profile", too, by default but can be changed to anything else).
- `Modal` content will now have its contents aligned to `start`, rather than `left`, to properly handle RTL direction.

### Fixed
- "Line Arrow" drawing tool would not save its color when serialized.
- "Line Arrow" tip was not taking a color change.
- ES2025 examples involving `MapChart` were not correctly importing geodata.
- Using `clearDrawings()` was erroring if there was "Line Arrow" drawings added to `StockChart`.
- Rotated elements with HTML content were not being properly positioned in all cases.
- `scrollToChild` method of `Container` was not working properly.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
